backfill/2023-06-16-mozillavpn-metrics-pings/README.md

```diff
@@ -0,0 +1,101 @@
+
+
+## Step 3:  Copy & deduplicate decoded `mozillavpn.metrics` pings
+
+To copy and deduplicate the rows from the "live" tables in the `moz-fx-data-backfill-3.mozillavpn_live` dataset into the "stable" tables in the `moz-fx-data-backfill-3.mozillavpn_stable` dataset, run the [`03-bigquery-etl-copy-deduplicate.sh` script](03-bigquery-etl-copy-deduplicate.sh):
+
+```bash
+cd path/to/bigquery-etl
+
+path/to/bigquery-backfill/backfill/2023-06-16-mozillavpn-metrics-pings/03-bigquery-etl-copy-deduplicate.sh
+```
+
+### Output (2023-06-16):
+  - 422,726 `moz-fx-data-backfill-3.mozillavpn_stable.metrics_v1` rows between 2022-12-20 and 2023-06-09.
+
+
+## Step 4:  Insert new `mozillavpn.metrics` pings and errors into production tables
+
+To insert the previously rejected `mozillavpn.metrics` pings and new decoder errors into production tables, someone with appropriate access can run the following `bq` commands either in Google Cloud Shell or using the Google Cloud SDK:
+
+```bash
+# The target live table's partition expiration is set to 30 days, so it's expected that most of the copied rows will immediately expire.
+# However, this is the easiest way to copy the data, and the data volume is very small so cost is not a factor.
+bq cp --append_table moz-fx-data-backfill-3:mozillavpn_live.metrics_v1 moz-fx-data-shared-prod:mozillavpn_live.metrics_v1
+
+bq cp --append_table moz-fx-data-backfill-3:mozillavpn_stable.metrics_v1 moz-fx-data-shared-prod:mozillavpn_stable.metrics_v1
+
+bq cp --append_table moz-fx-data-backfill-3:payload_bytes_error.structured moz-fx-data-shared-prod:payload_bytes_error.structured
+```
+
+### Output (2023-06-20):
+  - 422,868 `moz-fx-data-shared-prod.mozillavpn_live.metrics_v1` rows between 2022-12-22 and 2023-06-09.
+  - 422,673 `moz-fx-data-shared-prod.mozillavpn_stable.metrics_v1` rows between 2022-12-22 and 2023-06-09.
+  - 5,115 `moz-fx-data-shared-prod.payload_bytes_error.structured` rows.
+
+
+## Step 5:  Clean up
+
+1.  DE will remove any GCS buckets created by Dataflow jobs:
+
+    ```bash
+    # BE CAREFUL!  This removes all GCS buckets from the target project.  It cannot be undone.
+
+    gsutil ls -p moz-fx-data-backfill-3 | xargs gsutil -m rm -r
+    ```
+
+2.  Data SRE will remove the [resources that were created for this backfill with Terraform](https://github.com/mozilla-services/cloudops-infra/pull/4931) (note that this also removes DE editor access to the backfill GCP project).
+
+3.  Data SRE will delete the `mozillavpn.metrics` ping errors that have now been processed:
+
+    ```sql
+    DELETE FROM `moz-fx-data-shared-prod.payload_bytes_error.structured`
+    WHERE DATE(submission_timestamp) BETWEEN '2022-10-26' AND '2023-06-09'
+      AND document_namespace = 'mozillavpn'
+      AND document_type = 'metrics'
+      AND exception_class = 'com.mozilla.telemetry.ingestion.core.schema.SchemaNotFoundException'
+    ```
```

Comment:
`moz-fx-data-shared-prod:US.bquxjob_8b3d57d_188df13b6c3` in case you want to record the output (same  431,239 rows removed)